src/controllers/controller.bar.js

```diff
@@ -95,6 +95,92 @@ defaults._set('horizontalBar', {
 	}
 });
 
+/**
+ * Computes the "optimal" sample size to maintain bars equally sized while preventing overlap.
+ * @private
+ */
+function computeMinSampleSize(scale, pixels) {
+	var min = scale.isHorizontal() ? scale.width : scale.height;
+	var ticks = scale.getTicks();
+	var prev, curr, i, ilen;
+
+	for (i = 0, ilen = pixels.length; i < ilen; ++i) {
+		min = i > 0 ? Math.min(min, pixels[i] - pixels[i - 1]) : min;
+	}
+
+	for (i = 0, ilen = ticks.length; i < ilen; ++i) {
+		curr = scale.getPixelForTick(i);
+		min = i > 0 ? Math.min(min, curr - prev) : min;
+		prev = curr;
+	}
+
+	return min;
+}
+
+/**
+ * Computes the "ideal" sample range based on the absolute bar thickness or, if undefined or
```

Comment:
It would help to explain what a sample range is and potentially why there are different fit and flex sample ranges

Reply:
Relates to the description from the [docs](https://github.com/chartjs/Chart.js/blob/master/docs/charts/bar.md#barpercentage-vs-categorypercentage), but that's actually not "sample" but "category". It was initially a range, but after some refactor, it's not anymore one: I will rename these 2 methods `compute*CategoryTraits` (though, I don't like the "category" term here).